### DISCUSSION TITLE
forward_train() missing 1 required positional argument: 'gt_label'

### DISCUSSION THREAD

**Discussion by liuxy2318** [open-mmlab/mmclassification#1147] [Q&A]
When I run the mmclassification prediction demo, I get the correct prediction results, but when exporting the onnx model with torch.onnx.export(model, x, "resnet50.onnx", verbose=True), I always report an error. This error is the same as when I tried to export a model suitable for invocation in C++ using traced_module=torch.jit.trace(net, x).

**Comment by mzr1996** [comment]
Which version are you using?
If you are using mmcls 0.x, we have an onnx export script [here](https://github.com/open-mmlab/mmclassification/blob/master/tools/deployment/pytorch2onnx.py)
If you are using mmcls 1.x, we have moved our deploy-related functionality to [MMDeploy](https://mmdeploy.readthedocs.io/en/dev-1.x/04-supported-codebases/mmcls.html), welcome to use it!

**Reply by Songhuilun** [reply]
Hello，can you solve this problem?
